### PR TITLE
Create devboard accounts for developers and use it for assigning / creating tasks

### PR DESCRIPTION
**Pull request by markediez**

### CODE REVIEW

File: app/controllers/developers_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class DevelopersController < ApplicationController
   load_and_authorize_resource
   before_action :set_developer, only: [:show, :edit, :update, :destroy]
+  after_action :create_devboard_developer_account, only: [:create]
```
Comment (cthielen) [inline app/controllers/developers_controller.rb]: This should be in the model, not the controller. Use 'after_create' in the model.

File: app/models/developer.rb [hunk 2]
```diff
@@ -73,6 +73,18 @@ def assignments_at(date)
     # return valid_assignments
   end
 
+  def devboard_account
+    accounts.each do |account|
```
Comment (cthielen) [inline app/models/developer.rb]: A less expensive way to do this is something like: accounts.where() or accounts.find().

File: app/models/developer.rb [hunk 3]
```diff
@@ -73,6 +73,18 @@ def assignments_at(date)
...
+      return account if account.account_type == "devboard"
+    end
+  end
+
+  def devboard_account_id
```
Comment (cthielen) [inline app/models/developer.rb]: self.devboard_account().id is probably enough given that they have to have one.

File: app/models/developer.rb [hunk 4]
```diff
@@ -73,6 +73,18 @@ def assignments_at(date)
     # return valid_assignments
```
Comment (cthielen) [inline app/models/developer.rb]: Add a validation that a developer must have a DevBoard developer_account.

File: db/migrate/20170130214126_use_devboard_account_in_assignments.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class UseDevboardAccountInAssignments < ActiveRecord::Migration[5.0]
+  def change
+    Assignment.all.each do |a|
+      a.developer_account_id = a.developer.devboard_account_id
```
Comment (cthielen) [inline db/migrate/20170130214126_use_devboard_account_in_assignments.rb]: Do all assignments have developers?

Comment (markediez) [inline db/migrate/20170130214126_use_devboard_account_in_assignments.rb]: Hm. No. All assignments have developer accounts. Not all developer accounts are associated with a developer

**Pull request merged**